extra_data/writer.py

```diff
@@ -117,7 +117,8 @@ def write_metadata(self):
             metadata = {}
 
         metadata_grp = self.file.create_group('METADATA')
-        if metadata.get('dataFormatVersion') == '1.0':
+        format_version = float(metadata.get('dataFormatVersion'))
+        if format_version >= 1.0:
```

Comment:
```suggestion
        if metadata.get('dataFormatVersion') != '0.5':
```

The float comparison would work here, but we should avoid converting version numbers to floats just as a general thing, because version 1.10 is different from 1.1.

Reply:
The objects returned by `pkg_resources.parse_version` can be compared using inequality operators as expected for version numbers. I think we can assume `pkg_resources` to be available anyway with `setuptools`/`distutils` and it should not add a dependency.

Reply:
`pkg_resources` can be very slow to import, though - it scans all the other installed packages when it's imported. So I strongly prefer to avoid using it if at all possible.

I always used to use a LooseVersion class from distutils for this, but now distutils is deprecated - though we could still use it for now - I think it will be gone in Python 3.12 or something. Or we could do a very simple bit of parsing ourselves, like `[int(m) for m in re.findall('\d+', version)] .

So I stick to `==` and `!=` whenever possible :wink:

Reply:
TIL! Quick googling lead me to this: https://stackoverflow.com/a/11887885
Could that be an option? It seems to be installed in my `extra_data` dev environment without having done so manually, and I didn't notice any perceivable slowness when importing (unlike `pkg_resources` tested directly afterwards).

Reply:
Yup, good point, `packaging` is an option too. Technically it's a new dependency, but it's one that's pretty widely available.

Reply:
Yeouch good catch, I did not realize `float()` had that behaviour :fearful: I decided to go with `packaging` because it's also useful in the tests for 1.2 (3cb8ded). Fixed in c4f1cd7.